lib/puppet/functions/systemd/systemd_escape.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# @summary Escape strings by call the `systemd-escape` command in the background.
+Puppet::Functions.create_function(:'systemd::systemd_escape') do
+  # @param input Input string
+  # @param path Use path (-p) ornon-path  style escaping.
+  dispatch :escape do
+    param 'String', :input
+    optional_param 'Optional[Boolean]', :path
```

Comment:
I think optional booleans are odd. Can't we simply have a boolean with a default false? Or put in a different way: what's the difference here between `nil` / `undef` and `false`?

Reply:
I copy this from here https://github.com/puppetlabs/puppetlabs-stdlib/blob/276e7b7cd97580c9c08b264b3727e5902df1f487/lib/puppet/functions/to_json_pretty.rb#L48 where `Optional[Boolean]` is defined, but the default value is `false` (like line 14 here).